src/test/java/homework1/AccountServiceTest.java

```diff
@@ -0,0 +1,147 @@
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.Year;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AccountServiceTest {
+    AccountService accountService = new AccountService();
+
+    List<Account> accounts = List.of(
+            new Account("Pete", "Zet", "USA", LocalDate.of(1987, 3,12), 3600.00, "Male"),
+            new Account("Rose", "Love", "Canada", LocalDate.of(1990, 5,28), 13900.00, "Female"),
+            new Account("Grace", "Creek", "Great Britain", LocalDate.of(1999, 3,4), 600.00, "Female"),
+            new Account("Ron", "Smith", "Canada", LocalDate.of(1990, 5,21), 2400.00, "Male"),
+            new Account("Pride", "James", "Mexico", LocalDate.of(2000, 5,30), 10000.00, "Male"),
+            new Account("Hope", "Toronto", "USA", LocalDate.of(1979, 5,9), 5800.00, "Female"),
+            new Account("Liza", "Junior", "Great Britain", LocalDate.of(2000, 3,16), 1500.00, "Female")
+    );
+
+    @Test
+    void shouldOverCertainBalance() {
+
+    List<Account> expectedAccounts = List.of(
+            new Account("Rose", "Love", "Canada", LocalDate.of(1990, 5,28), 13900.00, "Female"),
+            new Account("Pride", "James", "Mexico", LocalDate.of(2000, 5,30), 10000.00, "Male"));
+
+            assertEquals(expectedAccounts,accountService.overCertainBalance(accounts, 10000.00));
+    }
+
+    @Test
+    void shouldListCountries() {
+
+        Set<String> countries = new HashSet<>(List.of("USA", "Canada", "Great Britain", "Mexico"));
+
+        assertEquals(countries, accountService.listCountries(accounts));
+    }
+
+    @Test
+    void shouldYoungerThanCertainYear() {
+
+        assertTrue(accountService.youngerThanCertainYear(accounts, 2000));
+        assertFalse(accountService.youngerThanCertainYear(accounts, 2006));
+    }
+
+    @Test
+    void shouldSumBalanceBeGender() {
+
+        assertEquals(16000.00, accountService.sumBalanceBeGender(accounts, "Male"));
+    }
+
+    @Test
+    void shouldGroupingByMonthBirth() {
+
+        Map<Month, List<Account>> map = new HashMap<>();
+
+        map.put(Month.of(3), List.of(
+                new Account("Pete", "Zet", "USA", LocalDate.of(1987, 3,12), 3600.00, "Male"),
+                new Account("Grace", "Creek", "Great Britain", LocalDate.of(1999, 3,4), 600.00, "Female"),
+                new Account("Liza", "Junior", "Great Britain", LocalDate.of(2000, 3,16), 1500.00, "Female")
+        ));
+        map.put(Month.of(5), List.of(
+                new Account("Rose", "Love", "Canada", LocalDate.of(1990, 5,28), 13900.00, "Female"),
+                new Account("Ron", "Smith", "Canada", LocalDate.of(1990, 5,21), 2400.00, "Male"),
+                new Account("Pride", "James", "Mexico", LocalDate.of(2000, 5,30), 10000.00, "Male"),
+                new Account("Hope", "Toronto", "USA", LocalDate.of(1979, 5,9), 5800.00, "Female")
+        ));
+
+        assertEquals(map, accountService.groupingByMonthBirth(accounts));
```

Comment:
comparing maps doesn't looks good, you can choose few different more narrow params